Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser.py

```diff
@@ -475,16 +553,130 @@ def assign_manager_command(client: MsGraphClient, args: dict):
 
+    return CommandResults(
+        outputs_key_field='ID',
+        readable_output=human_readable
+    )
+
+@suppress_errors_with_404_code
+def get_tap_policy_list_command(client: MsGraphClient, args: dict):
+    user_id = args.get('user_id')
+    policy_id = args.get('policy_id', None)
+    client_tap_data = client.list_tap_policy(user_id, policy_id)
+    tap_readable, tap_policy_output = parse_outputs(client_tap_data)
+    
+    # change HR from ID to Policy ID
+    if isinstance(tap_readable, list) and tap_readable:
+        tap_readable_dict = tap_readable[0]
+    else:
+        tap_readable_dict = tap_readable
+
+    if isinstance(tap_readable_dict, dict):
+        tap_readable_dict['Policy ID'] = tap_readable_dict.pop('ID', None)
+    headers = ['Policy ID', 'Start Date Time' , 'Lifetime In Minutes', 'Is Usable Once', 'Is Usable', 'Method Usability Reason']
+    
+    human_readable = tableToMarkdown(name='All TAP Policy Users', headers=headers, t=tap_readable_dict, removeNull=True)
+
+    return CommandResults(
+        outputs_prefix='MSGraphUser.TAPPolicy',
+        outputs_key_field='ID',
+        outputs=tap_policy_output,
+        readable_output=human_readable,
+        raw_response=client_tap_data
+    )
+
+@suppress_errors_with_404_code
+def create_tap_policy_command(client: MsGraphClient, args: dict):
+    user_id = args.get('user_id')
+    zip_password = args.get('zip_password')
+    lifetime_in_minutes = arg_to_number(args.get('lifetime_in_minutes'))
+    is_usable_once = argToBoolean(args.get('is_usable_once', False))
+    start_time = args.get('start_time', None)
+    start_time_iso = arg_to_datetime(start_time, required=False)
+    
+    fields = {
+    'lifetimeInMinutes': lifetime_in_minutes,
+    'isUsableOnce': is_usable_once,
+    'startDateTime': start_time_iso.strftime("%Y-%m-%dT%H:%M:%S.000Z") if start_time_iso is not None else None
+}
+    body = dict(fields.items())
+    res = client.create_tap_policy(user_id, body)
+    
+    if zip_password:
+        generated_password = res.get('temporaryAccessPass')
+        return_results(
+            create_zip_with_password(generated_password=generated_password, zip_password=zip_password)
+        )
+    human_readable = f'Temporary Access Pass Authentication methods policy {user_id} was successfully created'
+    _, tap_policy_output = parse_outputs(res)
+    
+    return CommandResults(
+        outputs_prefix='MSGraphUser.TAPPolicy',
+        outputs_key_field='ID',
+        outputs=tap_policy_output,
+        readable_output=human_readable
+    )
+
+@suppress_errors_with_404_code
+def delete_tap_policy_command(client: MsGraphClient, args: dict):
+    user_id = args.get('user_id')
+    policy_id = args.get('policy_id')
+    client.delete_tap_policy(user_id, policy_id)
+    human_readable = f'Temporary Access Pass Authentication methods policy {policy_id} was successfully deleted'
+    
+    return CommandResults(
+    outputs_key_field='ID',
```

Comment:
I don't think we need the outputs_key_field if we don't output anything to the context. 
But - we might want the raw response.